src/lib.rs

```diff
@@ -195,28 +200,45 @@ where
     pub fn from_slice(data: &'a [u8]) -> Result<Self, ParseError> {
         let raw = RawTga::from_slice(data)?;
 
-        Self::from_raw(raw)
-    }
-
-    /// Converts a raw TGA object into a embedded-graphics TGA object.
-    ///
-    /// # Errors
-    ///
-    /// If the bit depth of the source image does not match the bit depth of the output color type
-    /// `C`, this method will return a [`ParseError::MismatchedBpp`] error.
-    ///
-    /// [`ParseError::MismatchedBpp`]: enum.ParseError.html#variant.MismatchedBpp
-    pub fn from_raw(raw: RawTga<'a>) -> Result<Self, ParseError> {
-        if C::Raw::BITS_PER_PIXEL != usize::from(raw.color_bpp().bits()) {
-            return Err(ParseError::MismatchedBpp(raw.color_bpp().bits()));
-        }
+        let image_color_type = match (raw.color_bpp(), raw.image_type().is_monochrome()) {
+            (Bpp::Bits8, true) => ColorType::Gray8,
+            (Bpp::Bits16, false) => ColorType::Rgb555,
+            (Bpp::Bits24, false) => ColorType::Rgb888,
+            _ => {
+                return Err(ParseError::UnsupportedDynamicTgaType(
```

Comment:
More a note to self when I review this proper: This error variant would need renaming too.